utils/parsing.go

```diff
@@ -19,3 +21,87 @@ func ParseNumericArg(c *gin.Context, paramName string) (int, error) {
 		return numericId, nil
 	}
 }
+
+func ParseBigNumericArg(c *gin.Context, paramName string) (int64, error) {
+	// parse and validate:
+	numericArgValue := c.Param(paramName)
+	log.Printf("Querying %s: ", paramName)
+	if numericId, err := strconv.ParseInt(numericArgValue, 10, 64); err != nil {
+		log.Printf("bad request - %s should be a number", paramName)
+		return -1, fmt.Errorf("bad request - %s should be a number", paramName)
+	} else {
+		return numericId, nil
+	}
+}
+
+func ParseStringArg(c *gin.Context, paramName string) (string, error) {
+	// parse and validate:
+	stringArgValue := c.Param(paramName)
+	log.Printf("Querying %s: ", paramName)
+	if stringArgValue == "" {
+		log.Printf("bad request - %s should be set", paramName)
+		return "", fmt.Errorf("bad request - %s should set", paramName)
+	} else {
+		return stringArgValue, nil
+	}
+}
+
+func Pos(value int64, list []int64) int {
```

Comment:
always surprised that golang likes to really keep it simple sometimes, like too simple lol

Reply:
there must be something...right? This is too silly